### PR TITLE
Add replacement tx handling

### LINKED ISSUE #10: Speed up and replacement tx detection
**Problem:**
When a user triggers `speed-up-a-tx` or `cancel-a-tx` option in metamask, d0x dapps using this module are unable to detect the new state and hence all UI elements keep listening to old tx forever (until it's out of mempool?), practically requiring localstorage cleanup on user side.

**Solution:**
I've spent some time researching d0x modules related to the issue, and found out it's best to address it in this module in `::watch-events` as a hash-replacing fn. 
On web3 side the issue is well-known and being worked on (https://github.com/ChainSafe/web3.js/issues/3585#issuecomment-664624241), however as of now there is no separate event that web3 provider can notify us of (e.g. original tx hash listener resolving with failure).

I looked for existing solutions, IMO gnosis handled it nicely via https://github.com/gnosis/safe-react/pull/2473/files, there is this `findSpeedupTx` function that I started to implement; basically what it does is search pending block for tx by same nonce + sender and input data (if input data diverges it's a replacement tx, not a speed-up). 

`tl;dr` working on adding a `findSpeedupTx` fn to tx-watcher in this module to detect speed-up + replacement tx, based on a  gnosis PR

### PR DESCRIPTION
**Pull request by st4cks1defl0w**

### CODE REVIEW

**Comment by st4cks1defl0w** [comment]
@madvas r3ady for review. 
in short, i spent more time inspecting https://github.com/gnosis/safe-react/pull/2473/files solution, and storing any tx state is not necessary after all (purely based on receipt logic solution found in gnosis safe).

**Comment by st4cks1defl0w** [comment]
ok, commit history is horrible in this, one small commit is better IMO since changes are not that big

**Comment by st4cks1defl0w** [comment]
In order to benefit from the fix in d0x dapps after the merge, this dep can either be propagated via d0x tx lib (I can create a PR there to bump it) or as a direct dependency in dapps (with corresponding `:exclusion` in lein)

File: project.clj [hunk 1]
```diff
@@ -1,8 +1,9 @@
-(defproject district0x.re-frame/web3-fx "1.0.5"
+(defproject district0x.re-frame/web3-fx "1.0.6"
   :description "A re-frame effects handler for performing Ethereum Web3 API tasks"
   :url "https://github.com/district0x/re-frame-web3-fx"
   :license {:name "MIT"}
   :dependencies [[org.clojure/clojurescript "1.9.946"]
+                 [org.clojure/core.async "0.3.465"]
```
Comment (st4cks1defl0w) [inline project.clj:6]: avoiding epic "callback hell", since my changes required like two nested callbacks

**Review by madvas — APPROVED** [review]

**Pull request merged**